sopel/bot.py

```diff
@@ -592,6 +592,58 @@ def register_urls(self, urls: Iterable) -> None:
                 except plugins.exceptions.PluginError as err:
                     LOGGER.error("Cannot register URL callback: %s", err)
 
+    def rate_limit_info(self, rule: plugin_rules.Rule, trigger: Trigger) -> Tuple[bool, Optional[str]]:
+        if trigger.admin or rule.is_unblockable():
+            return False, None
+
+        is_channel = trigger.sender and not trigger.sender.is_nick()
+        channel = trigger.sender if is_channel else 'private-message'
```

Comment:
I don't like this. `channel` should be `None` here if it isn't a channel. And so is `channel_metrics`.

You could then replace the later `elif is_channel and [...]` by `elif channel_metrics is not None and [...]`.